dll/shellext/netshell/lanstatusui.cpp

```diff
@@ -992,108 +816,100 @@ CLanStatus::InitializeNetTaskbarNotifications()
        return S_OK;
     }
     /* get an instance to of IConnectionManager */
-
-    //hr = CoCreateInstance(&CLSID_ConnectionManager, NULL, CLSCTX_INPROC_SERVER, &IID_INetConnectionManager, (LPVOID*)&pNetConMan);
-
-    hr = INetConnectionManager_Constructor(NULL, IID_INetConnectionManager, (LPVOID*)&pNetConMan);
-    if (FAILED(hr))
-    {
-        ERR("INetConnectionManager_Constructor failed\n");
+    hr = CNetConnectionManager_CreateInstance(IID_PPV_ARG(INetConnectionManager, &pNetConMan));
+    if (FAILED_UNEXPECTEDLY(hr))
         return hr;
-    }
 
     hr = pNetConMan->EnumConnections(NCME_DEFAULT, &pEnumCon);
-    if (FAILED(hr))
-    {
-        ERR("EnumConnections failed\n");
-        pNetConMan->Release();
+    if (FAILED_UNEXPECTEDLY(hr))
         return hr;
-    }
 
     Index = 1;
-    do
+    while (TRUE)
     {
         hr = pEnumCon->Next(1, &pNetCon, &Count);
-        if (hr == S_OK)
+        if (hr != S_OK)
+            break;
+
+        TRACE("new connection\n");
+        pItem = static_cast<NOTIFICATION_ITEM*>(CoTaskMemAlloc(sizeof(NOTIFICATION_ITEM)));
```

Comment:
Why `CoTaskMemAlloc`?
Consider using `new` / `delete`, so that the object can clean up it's own resources.
It also removes the cast / `sizeof` stuff.